envparse.py

```diff
@@ -114,6 +114,22 @@ def __call__(self, var, default=NOTSET, cast=None, subcast=None,
         return value
 
     @classmethod
```

Comment:
I agree it's unlikely to be used elsewhere (and simply tracking it in the function was my first thought), but the `read_envfile` method [calls itself] (https://github.com/rconradharris/envparse/blob/master/envparse.py#L191-L192), so I used a classmethod to "log-sideways" as otherwise it would overwrite a var placed in the body.  The class method approach had a nifty little side effect of tracking the lookup attempts on each class, so I went that way.

The other way I thought to do this would be passing the "`traversed_paths`" along in the `read_envfile()` params.  I'l hack that up quickly for this PR.